### ISSUE TITLE
Debugging / analysis tool: Add ability to capture raw line protocol requests and replay them

### ISSUE THREAD

**Issue by alamb** [influxdata/influxdb_iox#1402]
Based on a conversation with @mkmik when we were talking about Chunk lifecycle

__Use case:__
* As an IOx developer  working to understand how the current lifecycle rules affect chunk movement and compression ratios 
* I want to be able to measure the effect various settings and code changes
* So that I can make data driven decisions on how to change the system and code

I would also like to create "real world" regression tests that send the same data through and then measure various performance metrics like "time to become readable", "time to persistence", "time for recovery", etc.

I would like to also be able to precisely calculate the actual compression ratio obtained by various systems. To do so I need to know, for the exact same data:
1. The size of the data as represented in line protocol
2. The size of the data as represented in a mutable buffer chunk
3. the size of the data in the read buffer
4. the size of the data in parquet file(s)

__Proposal:__
Add two new settings to the database rules: 
* `capture_requests`: {ALL, or list of [table_names]}
* `capture_requests_location`: 
* `max_capture_bytes`: uint64 (optional, defaults to 1GB)

When `capture_requests` is enabled, on every subsequent request to the http write endpoint, the following are captured
* time at which request was received
* http request headers
* relevant query parameters (e.g. org name, bucket name, etc)
* raw line protocol data (bytes, etc)
* (maybe) the actual time it took to process the request?

The data files are then written to `capture_requests_location` in a format (TBD) that can be replayed to another IOx instance.

Add a new `replay` command to `influxdb_iox`
* `influxdb_iox replay <capture_directory>` # replays the traffic linearly in order, as fast as possible 
* `influxdb_iox replay --timescale=wallclock <capture_directory>` # replays the traffic at the same speed as it was recorded

And probably other variants (e.g. write data using multiple concurrent requests)

__Current behaviour:__
We get metrics for parts of the data lifecycle but it is not possible (or very very hard) to correlate them together and it is hard to measure how particular code changes will affect the metrics without testing them

__Desired behaviour:__
The ability to replay traffic locally to speed up development time and make it more data driven

__Alternatives considered:__

Open questions:
1. Should the settings be environment configuration (e.g. `INFLUXDB_IOX_CAPTURE_BYTES` vs a database rule`)?
2. What format should be used (JSON? Lines of JSON? protobuf? flatbuffers?)
3. Should we write to local disk (probably?) or object store? Or allow the user to pick via config?

**Comment by mkmik** [comment]
I'm in two minds about adding this kind of "low level" things to the database config, which is effectively part of our API.
Perhaps an out-of-band config? OTOH, the database rules config is the simplest place to put stuff that is per db and that doesn't require the server to restart in order to pickup config changes.

**Comment by alamb** [comment]
Yeah -- I am not sure either (I need to let it marinate for a bit)

**Comment by NGA-TRAN** [comment]
I like the use cases a lot.
